jobs/distributable_lib.tcl

```diff
@@ -219,26 +220,34 @@ proc check_licenses {portName variantInfo} {
         set sub_names {}
         foreach full_lic $sublist {
             # chop off any trailing version number
-            set lic [remove_version [string tolower $full_lic]]
+            set lic [remove_version $full_lic]
             # add name to the list for later
             lappend sub_names $lic
-            if {[info exists ::license_good($lic)]} {
+            if {[info exists ::license_good([string tolower $lic])]} {
                 set any_good 1
             }
         }
         lappend top_license_names $sub_names
         if {!$any_good} {
-            return [list 1 "\"$portName\" is not distributable because its license \"$lic\" is not known to be distributable"]
+            lappend failures "\"$portName\" is not distributable because its license \"$lic\" is not known to be distributable"
         }
     }
 
     # start with deps of top-level port
+    set portPaths [dict create [lindex $top_info 0] [list]]
```

Comment:
Does this really need a dict rather than just an array? And what's the purpose of adding an entry for the whole list of dependencies?